src/ripple/consensus/Validations.h

```diff
@@ -0,0 +1,504 @@
+//------------------------------------------------------------------------------
+/*
+    This file is part of rippled: https://github.com/ripple/rippled
+    Copyright (c) 2012, 2013 Ripple Labs Inc.
+
+    Permission to use, copy, modify, and/or distribute this software for any
+    purpose  with  or without fee is hereby granted, provided that the above
+    copyright notice and this permission notice appear in all copies.
+
+    THE  SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+    WITH  REGARD  TO  THIS  SOFTWARE  INCLUDING  ALL  IMPLIED  WARRANTIES  OF
+    MERCHANTABILITY  AND  FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+    ANY  SPECIAL ,  DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+    WHATSOEVER  RESULTING  FROM  LOSS  OF USE, DATA OR PROFITS, WHETHER IN AN
+    ACTION  OF  CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+    OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+*/
+//==============================================================================
+
+#ifndef RIPPLE_CONSENSUS_VALIDATIONS_H_INCLUDED
+#define RIPPLE_CONSENSUS_VALIDATIONS_H_INCLUDED
+
+#include <ripple/basics/Log.h>
+#include <ripple/basics/UnorderedContainers.h>
+#include <ripple/basics/chrono.h>
+#include <ripple/beast/container/aged_container_utility.h>
+#include <ripple/beast/container/aged_unordered_map.h>
+#include <ripple/beast/utility/Journal.h>
+#include <ripple/beast/utility/Zero.h>
+#include <boost/optional.hpp>
+#include <utility>
+#include <vector>
+
+namespace ripple {
+
+/** Timing parameters to control validation staleness and expiration.
+ */
+struct ValidationParms
+{
+    /** The number of seconds a validation remains current after its ledger's
+        close time.
+
+        This is a safety to protect against very old validations and the time
+        it takes to adjust the close time accuracy window.
+    */
+    std::chrono::seconds VALIDATION_CURRENT_WALL = std::chrono::minutes{5};
```

Comment:
I think it depends on how you want folks to think about the values.  It sounds like the only reason they would change is for the simulation framework?  So do you want readers to think of these as though they are constants or as non-constants?  I suspect they are more likely to think of them as constants with names like`validationCURRENT_WALL`.  They are more likely to think of them as ordinary values with names like `validationCurrentWall`.  Naming conventions are mostly good for communicating intent to the reader/maintainer.  I guess I have a mild leaning toward `validationCurrentWall `, but I don't really have enough context for my opinion to carry much weight.